src/parser/grammar.lalrpop

```diff
@@ -863,15 +863,15 @@ NOpPre<ArgRule>: UniTerm = {
 }
 
 TypeBuiltin: Types = {
-    "Dyn" => Types(TypeF::Dyn),
-    "Num" => Types(TypeF::Num),
-    "Bool" => Types(TypeF::Bool),
-    "Str" => Types(TypeF::Str),
+    <l: @L> "Dyn" <r: @R> => Types { types: TypeF::Dyn, pos: mk_pos(src_id, l, r) },
```

Comment:
Those (and below) are still a lot of boilerplate position-related code leaking in the grammar. The idea of `WithPos` is to let `TypeBuiltin` as it is currently, but everywhere it's used, you use `WithPos<TypeBuiltin>` instead.